### PR TITLE
Fix platform wrapper tests

### PR DESCRIPTION
**Pull request by craig8**
# Description

Fixes a large number of stability issues with the testing framework.  

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tests are run against different combinations of actions.  The tests that are running are in the .github/workflows directory.  Currently the auth and platformwrapper tests are the two benchmark tests that are run using these.  I will be adding more, but want to make sure these base ones are complete before continuing.

### CODE REVIEW

File: volttrontesting/utils/platformwrapper.py [hunk 1]
```diff
@@ -1001,21 +1071,52 @@ def install_agent(self, agent_wheel=None, agent_dir=None, config_file=None,
...
+        if not self.is_running():
+            raise PlatformWrapperError("Instance must be running before starting agent")
 
+        self.__wait_for_control_connection_to_exit__(5)
+
+        cmd = [self.vctl_exe, '--json']
+        cmd.extend(['start', agent_uuid])
         result = execute_command(cmd, self.env)
 
+        self.__wait_for_control_connection_to_exit__(5)
```
Comment (schandrika) [inline volttrontesting/utils/platformwrapper.py:1110]: My understanding is that we are waiting here to make sure vctl start has completed all its work and relinquished its connection. But why should we wait before executing vctl start? Shouldn't we just wait after a vctl operations - start, stop, remove ? Do we need a wait after vctl status ?

File: volttrontesting/utils/platformwrapper.py [hunk 2]
```diff
@@ -1045,10 +1151,12 @@ def list_agents(self):
     def remove_agent(self, agent_uuid):
         """Remove the agent specified by agent_uuid"""
         _log.debug("REMOVING AGENT: {}".format(agent_uuid))
+        self.__wait_for_control_connection_to_exit__(5)
 
-        cmd = ['volttron-ctl']
+        cmd = [self.vctl_exe]
         cmd.extend(['remove', agent_uuid])
-        res = execute_command(cmd, env=self.env, logger=_log,
+        env = self.env.copy()
```
Comment (schandrika) [inline volttrontesting/utils/platformwrapper.py:1158]: why use a self.env.copy ?

File: volttrontesting/utils/platformwrapper.py [hunk 3]
```diff
@@ -565,18 +583,71 @@ def startup_platform(self, vip_address, auth_dict=None,
...
-        self.volttron_central_serverkey =volttron_central_serverkey
+        self.volttron_central_serverkey = volttron_central_serverkey
         self.bind_web_address = bind_web_address
+
+        if perform_preauth_service_agents:
+            authfile = AuthFile()
+
+            # pre-seed all of the volttron process identities before starting the platform
+            for identity in PROCESS_IDENTITIES:
+                capabilities = ['allow_auth_modifications']
```
Comment (schandrika) [inline volttrontesting/utils/platformwrapper.py:609]: where is this used? I currently see it only in test cases.

**Review by schandrika — CHANGES_REQUESTED** [review]

File: volttrontesting/utils/platformwrapper.py [hunk 4]
```diff
@@ -184,6 +186,10 @@ def create_volttron_home() -> str:
     # This is needed to run tests with volttron's secure mode. Without this
     # default permissions for folders under /tmp directory doesn't not have read or execute for group or others
     os.chmod(volttron_home, 0o755)
+    # Move volttron_home to be one level below the mkdir so that
```
[the remark this answers is not in the document]
Comment (craig8) [reply volttrontesting/utils/platformwrapper.py:189]: Thats part of the observer stuff.  it watches a folder and then shows us every time there is a modification to volttron.log...so its almost a recursive loop.

File: volttrontesting/utils/platformwrapper.py (hunk 3 above)
Comment (craig8) [reply volttrontesting/utils/platformwrapper.py:609]: this is what was there so I added it here..really need to look and make sure we have things that make sense for this

File: volttrontesting/utils/platformwrapper.py [hunk 5]
```diff
@@ -565,18 +583,71 @@ def startup_platform(self, vip_address, auth_dict=None,
...
         self.mode = mode
         self.volttron_central_address = volttron_central_address
-        self.volttron_central_serverkey =volttron_central_serverkey
+        self.volttron_central_serverkey = volttron_central_serverkey
         self.bind_web_address = bind_web_address
+
+        if perform_preauth_service_agents:
+            authfile = AuthFile()
+
+            # pre-seed all of the volttron process identities before starting the platform
```
[the remark this answers is not in the document]
Comment (craig8) [reply volttrontesting/utils/platformwrapper.py:607]: That's part of it.  
I saw a 3 second delay between when we did an update and when the auth service was able to pick things up and push them out to agents.

File: volttrontesting/testutils/test_single_instance.py [hunk 6]
```diff
@@ -8,7 +8,7 @@
 def test_can_install_listeners(volttron_instance):
```
[the remark this answers is not in the document]
Comment (craig8) [reply volttrontesting/testutils/test_single_instance.py:8]: ya true...will do

File: volttron/utils/__init__.py [hunk 7]
```diff
@@ -138,3 +138,15 @@ def on_any_event(self, event):
         except BaseException as e:
             _log.error("Exception in callback: {}".format(e))
         _log.debug("After callback on event {}".format(event))
+
+
+def print_stack():
```
[the remark this answers is not in the document]
Comment (craig8) [reply volttron/utils/__init__.py:143]: this isn't an exception this is I just want to see the stack of who called whom.

File: volttrontesting/testutils/test_platformwrapper.py [hunk 8]
```diff
@@ -349,51 +349,3 @@ def test_can_publish(volttron_instance):
 def test_fixture_returns_single_if_one_requested(get_volttron_instances):
     wrapper = get_volttron_instances(1, False)
     assert isinstance(wrapper, PlatformWrapper)
-
-
-@pytest.mark.wrapper
-def test_can_install_listener_on_two_platforms(get_volttron_instances):
```
[the remark this answers is not in the document]
Comment (craig8) [reply volttrontesting/testutils/test_platformwrapper.py:355]: done

File: volttrontesting/utils/platformwrapper.py (hunk 1 above)
Comment (craig8) [reply volttrontesting/utils/platformwrapper.py]: We need to wait for any calls that we shell out to vctl...each time we do that we have to wait until the previous exits.  that's why.  In general when running from command line it's not an issue but for this it is.  Besides having the wait here proves that we are ready to continue with execution.

File: volttrontesting/utils/platformwrapper.py (hunk 2 above)
Comment (craig8) [reply volttrontesting/utils/platformwrapper.py]: haha agreed...removed

File: volttrontesting/utils/platformwrapper.py (hunk 3 above)
Comment (craig8) [reply volttrontesting/utils/platformwrapper.py:656]: so I modified this code to only do it for masterweb....as @sgilbride  points out we need to update it when his pr comes in.

**Pull request merged**